language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/visitors/IsolatedBlockFinder.java

```diff
@@ -0,0 +1,56 @@
+package org.ballerinalang.langserver.command.visitors;
+
+import io.ballerina.compiler.syntax.tree.FunctionCallExpressionNode;
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.Token;
+
+import java.util.Optional;
+
+/**
+ * Finds if a provided {@link FunctionCallExpressionNode} node is contained within an isolated block.
+ *
+ * @since 2.0.0
+ */
+public class IsolatedBlockFinder extends NodeVisitor {
```

Comment:
Can't we use a node transformer here? Since we only capture the boolean status, it is better use a node transformer.

Reply:
+1 for that. But, we have to be aware of a possible NPE or CCE due to `ExternalTreeNodeList` returning null in it's `apply()` method. We cannot use primitive `boolean`. Instead, will have to use `Boolean` and handle `null` case as well.

Reply:
Done